Feature/cfox move tfservice from driver

Chris530906 opened the pull request:
Fixes from tag integration and moved attributes from context to new service object for real time acquisition of values.

Chris530906 committed:
added new tf service

new service which pulls down all attributes after terraform shell python package is intiated instead of getting passed a stale context. also refactored a bit to use new service and get attr instead of old way.

Chris530906 committed:
Moved tag manager to tf exec, fixed missing attribute bugs, cleaned up a few other areas

Chris530906 committed:
removing codys stuff

alexazarh commented on package/cloudshell/iac/terraform/terraform_shell.py:
```diff
@@ -20,14 +20,17 @@
 class TerraformShell:
     # todo: add support to provide the info needed from attributes as parameters to the init (not shell attributes)
-    def __init__(self, driver_context: ResourceCommandContext, terraform_service_shell: any,
+    def __init__(self, driver_context: ResourceCommandContext, cloudshell_model_name: str,
```
cloudshell_model_name -> we can get it from Context object

alexazarh commented on shells/generic_terraform_service/src/driver.py:
```diff
@@ -35,12 +35,12 @@ def execute_terraform(self, context: ResourceCommandContext):
         with LoggingSessionContext(context) as logger:
             tf_service = GenericTerraformService.create_from_context(context)
```
I think we can remove this, its not needed now because we can get the model name from the context

alexazarh commented on shells/generic_terraform_service/src/driver.py:
```diff
@@ -35,12 +35,12 @@ def execute_terraform(self, context: ResourceCommandContext):
 
     def destroy_terraform(self, context: ResourceCommandContext):
         with LoggingSessionContext(context) as logger:
             tf_service = GenericTerraformService.create_from_context(context)
```
I think we can remove this, its not needed now because we can get the model name from the context

alexazarh commented on package/cloudshell/iac/terraform/models/tf_service.py:
```diff
@@ -0,0 +1,23 @@
+    def set_context_resource_attributes(self) -> dict:
+        attr_dict = {}
+        services = self.api.GetReservationDetails(self.res_id).ReservationDescription.Services
+        for service in services:
+            for attribute in service.Attributes:
+                # self.context.resource.attributes[attribute.Name] = attribute.Value
+                attr_dict[attribute.Name] = attribute.Value
+        return attr_dict
```
I think there is a bug here, we need to get the attributes only for our service, not all services in the sandbox.
and also we can add a "break" if we find the correct service to stop the loop

Chris530906 replied:
Dan brought up the point there may be other services or cloudshell specific attributes we might want to access. Do we want to keep this list specific to this service?

Chris530906 replied:
done

alexazarh commented on package/cloudshell/iac/terraform/services/tf_proc_exec.py:
```diff
@@ -76,14 +75,15 @@ def destroy_terraform(self):
-            self._shell_helper.sandbox_messages.write_message("apply tags is false, skipping adding tags...")
-            return
+        apply = self._shell_helper.attr_handler.get_attribute(ATTRIBUTE_NAMES.APPLY_TAGS)
+        if apply:
+            if not strtobool(apply):
```
this can be merged into a single line

Chris530906 replied:
Please talk to me about this one

Chris530906 replied:
Nevermind will do the change.

alexazarh commented on package/cloudshell/iac/terraform/services/sandox_data.py:
```diff
@@ -21,7 +21,7 @@ def _get_tf_uuid(self) -> str:
             # Create new uuid and set it on the the attribute
             new_uuid = uuid.uuid4().hex
-            self._driver_helper_obj.tf_service.uuid = new_uuid
+            # self._driver_helper_obj.tf_service.uuid = new_uuid
```
I think we need to save the UUID in tf_service because its used later. @qualidan is it correct?

Chris530906 replied:
we are saving it in an attribute further down

Chris530906 committed:
fixes to pr

fixed temp dir deletion bug, cleaned up unneccassary objects, small logic changes

Chris530906 committed:
Merge branch 'master' into feature/cfox-move-tfservice-from-driver
